frontend/packages/kubevirt-plugin/src/components/modals/disk-modal/disk-modal.tsx

```diff
@@ -199,11 +199,15 @@ export const DiskModal = withHandlePromise((props: DiskModalProps) => {
   );
 
   const [accessMode, setAccessMode] = React.useState<AccessMode>(
-    isEditing ? (combinedDisk.getAccessModes() || [])[0] : applySP ? spAccessMode : null,
+    isEditing
+      ? (combinedDisk.getAccessModes() || [])[0]
+      : applySP && isSPSettingProvided
```

Comment:
nitpick: This combination of values is used four times in this block. In situations like this, I think it's better, if possible, to define a single variable with a descriptive name (maybe something like `applyProvidedSP`) and use it rather than repeating the combination. It improves readability, in my opinion.

Reply:
changed